tests/components/mikrotik/test_hub.py

```diff
@@ -88,9 +80,9 @@ def mock_command(self, cmd, params=None):
     # error when connection fails
     with patch(
         "librouteros.connect", side_effect=librouteros.exceptions.ConnectionError
-    ):
-        with pytest.raises(ConfigEntryNotReady):
-            await mikrotik.async_setup_entry(hass, CONFIG_ENTRY)
+    ), pytest.raises(ConfigEntryNotReady):
+        config_entry = MockConfigEntry(domain=mikrotik.DOMAIN, data=MOCK_DATA)
+        await mikrotik.async_setup_entry(hass, config_entry)
```

Comment:
This should change as above.

Reply:
I am trying `await hass.config_entries.async_setup(config_entry.entry_id)` but it won't work unless i do `config_entry.add_to_hass(hass)` which means the entry is added to hass and won't raise ConfigEntryNotReady.
Can you help me on this please.

Reply:
The `entry.state` will be set to `ENTRY_STATE_SETUP_ERROR` if the integration raises `ConfigEntryNotReady`.

https://github.com/home-assistant/home-assistant/blob/a99135a09e6ddfd2d9c05b0f9aa8e67c841a0ee4/homeassistant/config_entries.py#L223-L224

Reply:
Done